### PR TITLE
Support unembedded Tiled tilesets

### PR DESCRIPTION
**Pull request by fosskers**
The previous logic assumed that tilesets were embedded directly into the Tiled map. When handling many maps, it can be advantageous to leave the tileset unembedded so that its configuration can be more easily shared.

When embedded, the structure of the `tilesets` field of the original `.tmj` file looks like:
```json
[
    {
        "columns": 1,
        "firstgid": 1,
        "image": "ground.png",
        "imageheight": 16,
        "imagewidth": 16,
        "margin": 0,
        "name": "ground_tiles",
        "spacing": 0,
        "tilecount": 1,
        "tileheight": 16,
        "tilewidth": 16
    }
]
```
when unembedded, it looks like:
```json
[
    {
        "firstgid": 1,
        "source": "debugging.tsj"
    }
]
```
But luckily the format found in `.tsj` is (basically) identical to what the code originally expected, so we just need to do a quick switcheroo if the unembedded file is detected.

### CODE REVIEW

File: assets/tile-data.lisp [hunk 1]
```diff
@@ -55,20 +55,39 @@
...
+
+If the tileset is embedded, this yields NIL."
+  (let ((source (gethash "source" tileset)))
+    (when source
+      (let* ((as-path (pathname-utils:pathname* source))
+             (ext (pathname-type as-path)))
+        (if (or (string-equal "tsj" ext)
+                (string-equal "json" ext))
+            as-path
+            (error "Embedded tileset found, but not JSON format: ~a" as-path))))))
```
Comment (fosskers) [inline assets/tile-data.lisp:71]: Perhaps this function should be "pure" and simply yield what it found, and then the caller (`decode-tiled-tileset`) can be the one to raise the condition.

File: assets/tile-data.lisp [hunk 2]
```diff
@@ -55,20 +55,39 @@
...
-                                     :tile-size (vec (gethash "tileheight" data)
-                                                     (gethash "tilewidth" data)))
-        :first-id (gethash "firstgid" data)
-        :columns (gethash "columns" data)))
+(defun tileset-source (tileset)
+  "If the tileset is 'unembedded', yields the path to the .tsj/.json file where
+the actual tileset definition can be found. If the embedded tileset was saved in
+XML format, raise a condition.
+
+If the tileset is embedded, this yields NIL."
```
Comment (Shinmera) [inline assets/tile-data.lisp:63]: Please do not include docstrings in the source code, as per the [style guide](https://shirakumo.github.io/trial/style).

Especially a function as small as this is trivial to understand from the source.

**Comment by Shinmera** [comment]
While this lets you parse external tilesets, it doesn't actually solve the problem that they seek to solve. Namely, they way you implemented it here the tileset information is duplicated for each tilemap, meaning the tileset is loaded into a fresh texture for each tilemap and loaded anew for each as well, defeating the purpose of the deduplication in the first place.

In order to solve this properly we need a separate asset that holds the tileset information, and then make the tilemap figure out the corresponding asset for the tilemap file that it finds referenced in the json. How best to do that linkup I'm not sure.

**Comment by fosskers** [comment]
> the tileset information is duplicated for each tilemap, meaning the tileset is loaded into a fresh texture for each tilemap

As it would be for the behaviour on `master`, which assumes each map has its own embedded tileset. So you're right that this doesn't help deduplication within Trial itself, but as least it helps somewhat on the Tiled side. 

Provided I remove the docstring, we will be able to put this PR in as a first step toward an eventual complete external tileset solution?

> How best to do that linkup I'm not sure.

Same, as though as my understand of Trial grows I may think of something.

**Comment by Shinmera** [comment]
Sure

**Pull request merged**